### PR TITLE
Readme updates

### PR DESCRIPTION
**Pull request by mesmith027**
### Jira ticket 
https://fivetran.atlassian.net/browse/RD-1048879 

### Description of Change
Updates to the contributing section of the README for connector SDK. 

### Testing
readme compiles correctly and hyperlinks work 

### Checklist
Some tips and links to help validate your PR:

- [ NA] Tested the connector with `fivetran debug` command.
- [ NA] Added/Updated example specific README.md file, [refer here](https://github.com/fivetran/fivetran_connector_sdk/tree/main/template_example_connector/README_template.md) for template.
- [ NA] Followed Python Coding Standards, [refer here](https://github.com/fivetran/fivetran_connector_sdk/blob/main/PYTHON_CODING_STANDARDS.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -250,7 +250,11 @@ This repository is open source and intended specifically for Connector SDK examp
 2. Make your changes in a new branch: `git checkout -b feature/your-example-name`
 3. Add new connectors, fix bugs, improve documentation, or enhance existing features and commit your changes.
 4. Ensure your code works correctly and follows our coding standards.
+  * [Python coding standards](./PYTHON_CODING_STANDARDS.md)
+  * Fivetran coding principles
 5. Open a pull request with a clear description of your changes.
+  * If you're part of the AI Accelerate Google hackathon, please use the `accel Google hack 2025` tag
```
Comment (5tran-alexil) [inline README.md]: @alsnkline This tagging specifics is a temporal addition, correct? It will need to be removed after the hackathons?

File: README.md [hunk 2]
```diff
@@ -250,7 +250,11 @@ This repository is open source and intended specifically for Connector SDK examp
 2. Make your changes in a new branch: `git checkout -b feature/your-example-name`
 3. Add new connectors, fix bugs, improve documentation, or enhance existing features and commit your changes.
 4. Ensure your code works correctly and follows our coding standards.
+  * [Python coding standards](./PYTHON_CODING_STANDARDS.md)
+  * Fivetran coding principles
```
Comment (5tran-alexil) [inline README.md]:
```suggestion
  * Fivetran coding principles
```
@alsnkline These are only known to internal contributors, right? So I would not add this item to a doc that is consumed both internally and externally.

File: README.md [hunk 3]
```diff
@@ -250,7 +250,11 @@ This repository is open source and intended specifically for Connector SDK examp
 2. Make your changes in a new branch: `git checkout -b feature/your-example-name`
 3. Add new connectors, fix bugs, improve documentation, or enhance existing features and commit your changes.
 4. Ensure your code works correctly and follows our coding standards.
+  * [Python coding standards](./PYTHON_CODING_STANDARDS.md)
```
Comment (5tran-alexil) [inline README.md]: @mesmith027 Please change * to - in the bulleted lists for code consistency

File: README.md (hunk 1 above)
Comment (mesmith027) [reply README.md]: yes correct

File: README.md (hunk 2 above)
Comment (mesmith027) [reply README.md]: We are currently discussing the publication of an abbreviated guide for external contributors. This OS repo will exist for a long time, and we will expect external contributors to adjust their code to our standard for consistency. 

To reduce the overhead for our reviewers and the number of edits that an external contributor has to make, there needs to be a doc that we can point to externally to help guide them. 

To be clear, this PR will not be merged or ready until I have completed this doc and it is live, at which point we can put the link here.

Comment (alsnkline) [reply README.md]: Yes Marisa is flagging that as we move to having external contributors we need to make our code principles public - we were considering a new .md - or actually I think Varun added one yesterday. https://github.com/fivetran/fivetran_connector_sdk/blob/main/PYTHON_CODING_STANDARDS.md

File: README.md (hunk 3 above)
Comment (mesmith027) [reply README.md]: OK done, adjusted in [75ff975](https://github.com/fivetran/fivetran_connector_sdk/pull/309/commits/75ff9756a026246bc3d553024a0b394c97cf079d)

@alsnkline: A perfect example of something we can add to the FT coding standards doc

File: README.md (hunk 2 above)
Comment (mesmith027) [reply README.md]: resolved with https://github.com/fivetran/fivetran_connector_sdk/pull/309/commits/75ff9756a026246bc3d553024a0b394c97cf079d

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
+# Fivetran code review and example connector guidelines
+
+---
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]:
```suggestion
```

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
+# Fivetran code review and example connector guidelines
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]:
```suggestion
# Fivetran Code Review and Example Connector Guidelines
```

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+
+
+
+### 5.3 Avoid overloading
+
+- Don’t demonstrate all features in one example.
+- Create separate folders/examples, each with a distinct goal.
+- Keep the `update()` method short — use smaller helper methods.
+
+---
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]:
```suggestion
```

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+These guidelines ensure all Fivetran examples and code reviews are:
+- High quality.
+- Easy to understand.
+- Consistent with best practices.
+- Collaborative and maintainable.
+
+Following these principles leads to faster reviews, cleaner merges, and more resilient code.
+
+---
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]:
```suggestion
```

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+
+    where `type_of_change` is one of:  `feature`, `fix`, `test`, `cleanup`, `refactor`, `doc`, `flag`, `tweak`, `config`, `security`.
+
+    Examples:
+    - `feature(examples): added new example for smartsheets  ## New example`
+    - `fix(examples): fixes response parsing in github example  ## Fix existing example`
+    - `fix(source_examples): added comments in the examples  ## Fix multiple existing examples`
+
+2. Write a detailed PR description, including:
+- Links to relevant tickets.
```
Comment (alsnkline) [inline FIVETRAN_CODING_PRINCIPLES.md:46]: should this be issues?

we don't have a public ticketing system but are planning on using github issues for this in context of the repo.

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+- How the change was tested.
+- Any additional context useful to reviewers.
+
+3. Review your own code before submission:
+- Validate naming conventions, boundary conditions, and test coverage.
+- Feel free to use GitHub Copilot as a reviewer to catch basic issues early.
```
Comment (alsnkline) [inline FIVETRAN_CODING_PRINCIPLES.md:52]: should we mention that we  have an automated python standards linter configured and you should address issues it finds ideally before requesting human review.

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 10]
```diff
@@ -0,0 +1,192 @@
...
+5. Respond professionally to reviewer comments. Remember we are here to help and will strive to ask for only necessary changes to future proof your contribution and make it understandable to others.
+
+---
+
+## 3. Preparing code for PR submission
+
+To adhere to Fivetran engineering standards, follow this process:
+
+1. Write your `connector.py` solution (start from system-generated baselines if applicable).
+2. Feed it to the cursor agent `connector_sdk_template` to generate `connector_test.py` following best practices.
```
Comment (alsnkline) [inline FIVETRAN_CODING_PRINCIPLES.md:65]: Is this really a 'cursor agent'?

what happens if I don't have cursor
I'm also not sure I could follow these instructions - what is the cursor agent? - where do I find it?

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 11]
```diff
@@ -0,0 +1,192 @@
...
+3. Review generated revisions (e.g., ensure `fivetran debug` works).
+4. Feed the project to cursor agent `connector_sdk_readme` to generate a consistent `README.md`.
```
Comment (alsnkline) [inline FIVETRAN_CODING_PRINCIPLES.md:67]: similar concerns/questions about 'cursor agent'

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 12]
```diff
@@ -0,0 +1,192 @@
...
+- Add docstrings explaining each function’s purpose.
+- Add comments for complex logic and user customization points.
+- Split large functions into smaller helper functions for readability.
+- Use logging judiciously — avoid excessive or redundant logs.
+- Implement error handling that targets specific exceptions.
+- Add retry logic (with exponential backoff) for API requests.
+- Define a complete data model in the `schema()` function, with primary keys and data types.
+- Avoid loading all data into memory — use pagination or streaming.
+- Document pagination, upsert/update/delete, and checkpointing with clear comments.
+- Checkpoint state regularly to resume from the last successful sync.
```
Comment (alsnkline) [inline FIVETRAN_CODING_PRINCIPLES.md]:
```suggestion
- Checkpoint state regularly to resume from the last successful sync and cleanly send data to your warehouse.
```

Checkpoint has a subtle but important role in that is also tell the system that this batch of data is coherent and can be delivered. Even if you are fully refreshing your tables every sync and so don't really need to manage state it is good practice to periodically checkpoint.

**Review by alsnkline — APPROVED** [review]
I did find a few things - although we can iterate after shipping also. I don't want to hold this up.

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 8 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md:46]: totally can be

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 9 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md:52]: thats great yes

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 11 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md:67]: Honestly, I'm glad you feel this way. Since I am so new I thought i just didn't know FT tech well enough to know what we were trying to convey here (and figured I could gain knowledge and make some clarification edits later). 

I am now thinking that its confusing and also a bit redundnt given a simpler version of this is already on the README. I think we jsut axe this section. Will do that now.

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 8 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md:46]: addressed in #309

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 9 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md:52]: addressed in #309

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 10 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md:65]: addressed in #309

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 11 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md:67]: addressed in #309

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 13]
```diff
@@ -0,0 +1,192 @@
...
+5. Submit the PR.
+
+---
+
+## 4. Example connector guidelines
+
+### 4.1 Template connector
+
+A reference implementation is available here:  
+🔗 [Fivetran Connector SDK Template Example](https://github.com/fivetran/fivetran_connector_sdk/tree/main/template_example_connector)
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]:
```suggestion
[Fivetran Connector SDK Template Example](https://github.com/fivetran/fivetran_connector_sdk/tree/main/template_example_connector)
```
No emojis

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 14]
```diff
@@ -0,0 +1,192 @@
...
+- Self-explanatory and well-commented code.
+- Must produce working code (can require the SDK Playground).
+- No identifiable customer information should be included.
+
+
+
+### 5.2 Demonstrate one core concept
+
+- Each example should focus on one SDK feature or pattern (e.g., incremental sync, schema discovery, error handling).
+- Split complex connectors into multiple examples.
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]: @mesmith027 I am not sure I am following what it means. A complex connector is one that has many features. For source connectors, it's unfeasible to "split", a connector works as a whole and single entity. For pattern/feature implementation examples, it's an inherent and obvious core principle to implement that very feature and nothing else.

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 15]
```diff
@@ -0,0 +1,192 @@
...
+
+- Don’t demonstrate all features in one example.
+- Create separate folders/examples, each with a distinct goal.
+- Keep the `update()` method short — use smaller helper methods.
+
+
+### 5.4 Use a clear, self-contained folder
+
+- Use descriptive names like `incremental_sync_example` or `dynamic_schema_example`.
+- Include a simple `README.md` describing:
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]: @mesmith027 We have more than general 2 rules for source connector example readmes - we have a [detailed template](https://github.com/fivetran/fivetran_connector_sdk/blob/main/template_example_connector/README_template.md) that the contributor is expected to follow. We may need to outline these 2 types and for source connectors, link to the readme template.

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 16]
```diff
@@ -0,0 +1,192 @@
...
+### 5.5 Minimal dependencies
+
+- Include only necessary dependencies.
+- Pin dependency versions in `requirements.txt`, `pyproject.toml`, or `poetry.lock`.
+
+
+### 5.6 Code organization
+
+- Keep the code easy to read and navigate.
+- Use good naming instead of excessive commenting.
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md:142]: I don't think it's either or. Good naming and detailed commenting is what works. LongNamesDescribingImplementationDetailsOfIncrementalSyncsForCaseWhereTableHasNoPrimaryKeys does not replace comments, excessive or not.

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 17]
```diff
@@ -0,0 +1,192 @@
...
+- Follow established commenting conventions:
+  - Explain the example’s purpose.
+  - Explain helper functions with inline comments.
+  - Avoid monolithic `update` functions — split logic into helpers.
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]: Are we talking about Update method? We need to be specific since we have this naming collision Update methos and update operation. Of course, an experienced user (of CSDK) may conclude from the context it's about the method, yet we want the documentation also be clear for newcomers. 
@alsnkline what was the reason behind this? I found this confusing when I started looking into CSDK.

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 18]
```diff
@@ -0,0 +1,192 @@
...
+
+
+
+### 5.7 Logging
+
+- Log key steps at info level (e.g., `"Fetching page X"`).
+- Avoid logs inside loops or spammy debug logs.
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]: "spammy debug logs"
Sounds very vague to me without criteria. Plus whoever will be using the working example as is should not have issues at debugging/testing. And whoever will be modifying the connector should be able to clean up "spammy" logs as they find fit.

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 19]
```diff
@@ -0,0 +1,192 @@
...
+- Add logs for warnings and severe issues.
+
+
+### 5.8 Error handling
+
+- Catch and raise meaningful, specific exceptions.
+- Avoid generic exception handling.
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]: "generic" as in "generic Python" exception handling? Like, do we recommend implementing custom handling for everything?

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 20]
```diff
@@ -0,0 +1,192 @@
...
+### 5.9 Testing (optional but encouraged)
+
+- Include a basic test file (e.g., `test_connector.py`).
+- Use `pytest` or similar for validation.
+- Mock external API calls so examples run without real credentials.
+
+### 5.10 Document known limitations
+
+- Note any unhandled edge cases.
+- Provide guidance for contributors on how to extend or enhance the example.
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]: IMHO, this recommendation is too challenging unless the author has some improvement on their mind but didn't implement it (due to what reasons?). I can't imagine such a case being common.

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 21]
```diff
@@ -0,0 +1,192 @@
...
+
+---
+
+## 6. Summary
+
+These guidelines ensure all Fivetran examples and code reviews are:
+- High quality.
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]:
```suggestion
- High quality
```

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 22]
```diff
@@ -0,0 +1,192 @@
...
+- Easy to understand.
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]:
```suggestion
- Easy to understand
```

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 23]
```diff
@@ -0,0 +1,192 @@
...
+- Consistent with best practices.
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]:
```suggestion
- Consistent with best practices
```

File: FIVETRAN_CODING_PRINCIPLES.md [hunk 24]
```diff
@@ -0,0 +1,192 @@
...
+- Collaborative and maintainable.
```
Comment (5tran-alexil) [inline FIVETRAN_CODING_PRINCIPLES.md]:
```suggestion
- Collaborative and maintainable
```

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 20 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md]: kk I will axe it

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 19 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md]: TBH i think this is a bit repetitive of the line above going to axe it

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 16 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md:142]: Its not. it's a balance of both. will address

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 17 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md]: Oh, I didn't know this. If there is ambiguity, we should remove this and we could address it down the line if needed

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 18 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md]: will remove

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 15 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md]: Hmm, I already linked to that template, and this is feeling repetitive. will remove and reorder

File: FIVETRAN_CODING_PRINCIPLES.md (hunk 14 above)
Comment (mesmith027) [reply FIVETRAN_CODING_PRINCIPLES.md]: Leaving a message for you in our open Slack convo about this one, as it links to some internal documentation that I am unsure if I can post publicly.

**Review by 5tran-alexil — APPROVED** [review]
LGTM
